src/utils/hooks/useSetDefaultTicketSearchParams.ts

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import { useQuery } from '@tanstack/react-query';
+import { fetchOrganizations } from '../../api/cristinApi';
+import { fetchUser } from '../../api/roleApi';
+import { TicketSearchParam } from '../../api/searchApi';
+import { RootState } from '../../redux/store';
+import { TicketStatus } from '../../types/publication_types/ticket.types';
+import { Organization } from '../../types/organization.types';
+
+function flattenOrganizationsWithSubunits(topLevelOrganizations: Organization[]): Organization[] {
+  return topLevelOrganizations.flatMap((org) => flattenOrganizationWithSubunits(org));
+}
+
+function flattenOrganizationWithSubunits(org: Organization): Organization[] {
+  const subUnits = org.hasPart?.flatMap((subOrg) => flattenOrganizationWithSubunits(subOrg)) || [];
+  return [org, ...subUnits];
+}
+
+export const useSetDefaultTicketSearchParams = (): boolean => {
+  const [defaultParamsAdded, setDefaultParamsAdded] = useState(false);
+
+  const { t } = useTranslation();
+  const user = useSelector((store: RootState) => store.user);
+  const nvaUsername = user?.nvaUsername ?? '';
+
+  const history = useHistory();
+
+  const institutionUserQuery = useQuery({
+    enabled: !!nvaUsername,
+    queryKey: ['user', nvaUsername],
+    queryFn: () => fetchUser(nvaUsername),
+    meta: { errorMessage: t('feedback.error.get_person') },
+  });
+  const areasOfResponsibilityIds = institutionUserQuery.data?.viewingScope?.includedUnits ?? [];
+
+  const organizationQuery = useQuery({
+    enabled: areasOfResponsibilityIds.length > 0,
+    queryKey: ['organizations', areasOfResponsibilityIds],
+    queryFn: async () => fetchOrganizations(areasOfResponsibilityIds),
+    meta: { errorMessage: t('feedback.error.get_institution') },
+  });
+
+  useEffect(() => {
+    if (!organizationQuery.data) {
+      return;
+    }
+
+    const searchParams = new URLSearchParams(history.location.search);
+
+    const organizations = organizationQuery.data;
+    const organizationsWithSubUnits = flattenOrganizationsWithSubunits(organizations);
```

Comment:
Ser ut som funksjonen vi har kan brukes som direkte erstatning faktisk:
```suggestion
    const organizationsWithSubUnits = getSortedSubUnits(organizations);
```

Kan vurdere hva som er verdt å gjenbrukes her. Om vi skal gjenbruke denne burde vi kanskje rename den til `getAllUnits` elns

Reply:
Istedenfor å rename endte jeg opp med å bruke `getAllChildOrganizations()` direkte istedenfor å gå via `getSortedSubUnits()`, ettersom jeg ikke bryr meg om rekkefølgen.